src/core/primitives.rs

```diff
@@ -291,20 +293,14 @@ pub fn update(
             let h_i = &h * x_i;
             // Set the correct hybridization property.
             let (sk_i, pk_i) = if is_hybridized == EncryptionHint::Hybridized {
-                let (pk_i, _) = mpk.subkeys.get(partition).ok_or_else(|| {
-                    Error::KeyError(
-                        "Kyber public key cannot be computed from the secret key.".to_string(),
-                    )
-                })?;
+                let pk_i = mpk
+                    .subkeys
+                    .get(partition)
+                    .map(|(pk_i, _)| pk_i)
+                    .unwrap_or(&None);
```

Comment:
You can compute this in the if sk_i.is_some() {...} branch to avoid needless computation.